### PR TITLE
#13 - Network Configuration.

### LINKED ISSUE #13: Different configuation for testnet, local and mainnet
## Description:
Maintain different Ceramic, Threads, contract, and other endpoints for testnet, local, and mainnet config.

## Things to do:
Maintain the endpoints and values similar to the network config on `contracts-js` but with only 3 keys (testnet, local and mainnet)

### PR DESCRIPTION
**Pull request by yathishram**
## Description:
This PR adds a network configuration for each network.

## Added:

List the added features
* Networks JSON file where each network has `chainId`, `addresses`, `threadId`.

## Modified:

List the modified features
* Constructor - Constructor takes `signer`, `network`, `databaseType`, `APIkey`, `APISecret`, `threadId`

## Deleted:

List the modified features
* Removed parameters for `loginUser`

### CODE REVIEW

File: src/SafientCore.ts [hunk 1]
```diff
@@ -41,30 +40,50 @@ export class SafientCore {
...
-   * @param databaseType - Type of Database being used.
+   * @param signer - Signer object of the wallet
+   * @param network - The type of network from NetworkType Enum
+   * @param databaseType - Type of database to use.
+   * @param databaseAPIKey - Database API key.
+   * @param databaseAPISecret - Database API secret.
+   * @param threadId - ThreadDB ID if its available (optional)
    */
-  constructor(signer: Signer, chainId: number, databaseType: string) {
+  constructor(signer: Signer, network: number, databaseType: string, databaseAPIKey: any, databaseAPISecret: any, threadId: number[] | null ) {
```
Comment (koshikraj) [inline src/SafientCore.ts]: Change the network type from `number` to `NetworkType`

File: src/types/types.ts [hunk 2]
```diff
@@ -185,4 +185,47 @@ export interface GenericSafe {
 
 export type SafeStore = {
   safe: CryptoSafe | GenericSafe
+}
+
+export type  Networks = {
```
Comment (koshikraj) [inline src/types/types.ts]: Remove this if not used.

File: src/lib/enums.js [hunk 3]
```diff
@@ -13,7 +13,11 @@ export const claimStages = {
       "REJECTED": 3
   }
 
-  export const SafeType = {
-    "CRYPTO": 0,
-    "GENERIC": 1
-  }
\ No newline at end of file
+  export const NetworkType = {
```
Comment (koshikraj) [inline src/lib/enums.js]: Let's refactor all the enums in this file. We do not need to define the value for enums. This one should be:
```
export enum NetworkType {
mainnet,
 testnet,
 local,
devnet
}
```
Use all small for the enum values. And network type should have only these values.

Refactor all the enums in this file and maintain the naming consistency.

File: test/scenario1.js [hunk 4]
```diff
@@ -48,7 +48,7 @@ describe('Scenario 1 - Creating safe offChain', async () => {
   })
   //Step 1: Register all users
   it('Should register a Creator', async () => {
-    creatorSc = new SafientCore(creatorSigner, Enums.NetworkType.LOCALHOST, 'threadDB', apiKey, secret, null);
```
Comment (koshikraj) [inline test/scenario1.js]: You haven't updated the case in the enum file. Tests are failing.

Comment (yathishram) [reply test/scenario1.js]: @koshikraj this is outdated code. The new code has updated enum keys. Look into it once.

**Review by koshikraj — APPROVED** [review]

**Pull request merged**